### PR TITLE
Feature/fwm 15/pilot functions

### PR DESCRIPTION
**Pull request by PAKovalev**
#FMW-15 #FMW-3

Pilot and passenger functions for fly with me.

[https://momentum.nifty.pm/l/PCaWuUXuDV5c](https://momentum.nifty.pm/l/PCaWuUXuDV5c)
[https://momentum.nifty.pm/l/LFzUkJKypB](https://momentum.nifty.pm/l/LFzUkJKypB)

### CODE REVIEW

File: packages/app/src/shared/services/i18n/locales/en-gb.ts [hunk 1]
```diff
@@ -330,7 +331,11 @@ export const enGb = {
         'This is the Momentum world builder - a place where you can create your own decentralised social metaverse. Take ownership of your own world to build and create together. Shape your metaverse to suit your needs, enable the functionality you want to use and invite the people you want to collaborate with.',
       startBuilding: 'Connect your wallet\n To start Building',
       savingSpaceDetailsSuceess: 'Space details saved succesfully!',
-      stageIsFullTryAgain: 'Stage is full, please try again later...'
+      stageIsFullTryAgain: 'Stage is full, please try again later...',
+      flyWithEnabled: 'Fly with me enabled',
+      flyWithActivated: 'Fly with me has been activated',
+      flyWithDisabled: 'Fly with me has been disabled',
+      flyWithBlocked: 'While in fly with me mode this functionality is blocked.'
```
Comment (Daboo1998) [inline packages/app/src/shared/services/i18n/locales/en-gb.ts:338]: I guess it should be flyWithMe... not flyWith...

**Review by Daboo1998 — APPROVED** [review]

File: packages/app/src/api/repositories/flyWithMeRepository/flyWithMeRepository.api.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const url = generatePath(flyWithMeRepositoryEndpoints().start, {spaceId});
+  return request.post(url, {}, restOptions);
+};
+
+export const stop: RequestInterface<StopFlyWithMeRequest, void> = (options) => {
+  const {spaceId, ...restOptions} = options;
+
+  const url = generatePath(flyWithMeRepositoryEndpoints().stop, {spaceId});
+  return request.post(url, {}, restOptions);
+};
```
Comment (dmitry-yudakov) [inline packages/app/src/api/repositories/flyWithMeRepository/flyWithMeRepository.api.ts:21]: Offtop: 
Is it just me or you guys also copypaste these from some other file? :)
Should we think of some reusable pieces of logic for such calls or it's too much?

File: packages/app/src/scenes/flight/pages/FlyWithMePage/FlyWithMePage.tsx [hunk 3]
```diff
@@ -1,34 +1,82 @@
...
-    flyWithMeStore.start();
-    unityStore.toggleMiniMap();
+    if (!agoraStore.hasJoined) {
+      history.push(ROUTES.base);
+    }
+  }, [agoraStore, history]);
+
+  const isPilot = useMemo(() => {
+    return sessionStore.userId === pilotId;
+  }, [pilotId, sessionStore.userId]);
```
Comment (dmitry-yudakov) [inline packages/app/src/scenes/flight/pages/FlyWithMePage/FlyWithMePage.tsx:32]: It's just one comparison and boolean result - no point of memoizing.

File: packages/app/src/api/repositories/flyWithMeRepository/flyWithMeRepository.api.ts (hunk 2 above)
Comment (PAKovalev) [reply packages/app/src/api/repositories/flyWithMeRepository/flyWithMeRepository.api.ts:21]: Maybe, but now it is quite generic.

**Review by dmitry-yudakov — APPROVED** [review]
approved with small remark

File: packages/app/src/scenes/flight/pages/FlyWithMePage/FlyWithMePage.tsx (hunk 3 above)
Comment (PAKovalev) [reply packages/app/src/scenes/flight/pages/FlyWithMePage/FlyWithMePage.tsx:32]: It will be executed only once, not for each re-render.

Comment (PAKovalev) [reply packages/app/src/scenes/flight/pages/FlyWithMePage/FlyWithMePage.tsx:32]: But basically, yes. It doesn't make a big sense.

**Pull request merged**